### ISSUE TITLE
OS X install broken: no webkitgtk for webkit1 available

### ISSUE THREAD

**Issue by liljenstolpe** [astroidmail/astroid#235]
After installing the prerequisites as identified in the OS X page in the wiki, scons errors not being able to find webkitgtk.

However, it is installed, per brew info webkitgtk.

Thoughts?

**Comment by gauteh** [comment]
Hi, the problem is that you need the older line of webkit - which I have not yet been able to find a proper webkitgtk homebrew package for yet. There exists some, but I haven't been able to compile them.

Depends on #49: Port to newer webkitgtk.

**Comment by gauteh** [comment]
Basically, if you can find `webkitgtk-2.4.11` then it should support webkit1.

**Comment by liljenstolpe** [comment]
I'll look, thx.

**Comment by hugoroy** [comment]
I'm interested to know as well how it went @liljenstolpe  :-)

**Comment by liljenstolpe** [comment]
Copious spare time fail :)

**Comment by hugoroy** [comment]
Any mac user interested in this? Having astroid in homebrew would maybe help get new users and contributors :-) cc @liljenstolpe

**Comment by cpiro** [comment]
It was painful, but I got `brew` to build `webkitgtk-2.4.11` on macOS Sierra: [cpiro/webkitgtk@2.4.11.rb](https://gist.github.com/cpiro/0ea18227b251b26f94c645dd62b3f9a7)

I have no idea whether it's configured with enough features to run astroid, but it's a start. I'm in the process of getting astroid itself to build; hopefully I'll have something to show for it soon.

**Comment by gauteh** [comment]
Chris Piro writes on mars 10, 2017 7:15:
> It was painful, but I got `brew` to build `webkitgtk-2.4.11` on macOS Sierra: [cpiro/webkitgtk@2.4.11.rb](https://gist.github.com/cpiro/0ea18227b251b26f94c645dd62b3f9a7)
> 
> I have no idea whether it's configured with enough features to run astroid, but it's a start. I'm in the process of getting astroid itself to build; hopefully I'll have something to show for it soon.
> 

This is great news! Let me know if you run into other issues, I'm sure
they can be resolved somehow. Hopefully this can end up as an Mac OS X
section in the wiki along with other distributions.

**Comment by gauteh** [comment]
@cpiro: any progress? did you manage to compile astroid?

**Comment by cpiro** [comment]
The good news is that I got past the config step on to compilation: [cpiro/astroid.rb](https://gist.github.com/cpiro/dcc8ae89322250bf9379201b30a6a377). The bad news is that `gtkmm/socket.h` is unavailable on macOS, at least from the `gtkmm3` Homebrew formula built from source and poured from bottle. From `gtkmm/socket.h`:

```
//This file is used only on systems with the X11 backend.
//This is available only #ifdef GDK_WINDOWING_X11.
//Otherwise this source file will not be compiled,
//and no class defined in it will be registered by wrap_init().
```

I don't know much about building software on macOS with X available _at build time_, but it sounds like even less fun than above. The headers are around on my system though – I have XQuartz installed – so I'll give it a whirl.

**Comment by gauteh** [comment]
Chris Piro writes on mars 27, 2017 22:41:
> The good news is that I got past the config step on to compilation: [cpiro/astroid.rb](https://gist.github.com/cpiro/dcc8ae89322250bf9379201b30a6a377). The bad news is that `gtkmm/socket.h` is unavailable on macOS, at least from the `gtkmm3` Homebrew formula built from source and poured from bottle. From `gtkmm/socket.h`:
> 
> ```
> //This file is used only on systems with the X11 backend.
> //This is available only #ifdef GDK_WINDOWING_X11.
> //Otherwise this source file will not be compiled,
> //and no class defined in it will be registered by wrap_init().
> ```
> 
> I don't know much about building software on macOS with X available _at build time_, but it sounds like even less fun than above. The headers are around on my system though – I have XQuartz installed – so I'll give it a whirl.
> 
> -- 

Ah, this can be avoided by only allowing external editors. But, it
requires some compile-time definitions.

**Comment by cpiro** [comment]
I should also mention that it took me like 15 minutes to get astroid running in (Docker for Mac)[https://docs.docker.com/docker-for-mac/install/]. I started with Ubuntu 16.10, followed your install instructions, then `ssh -YC`'d into the container. This might be the easier path.

**Comment by gauteh** [comment]
Gaute Hope writes on mars 27, 2017 22:44:
> Chris Piro writes on mars 27, 2017 22:41:
>> The good news is that I got past the config step on to compilation: [cpiro/astroid.rb](https://gist.github.com/cpiro/dcc8ae89322250bf9379201b30a6a377). The bad news is that `gtkmm/socket.h` is unavailable on macOS, at least from the `gtkmm3` Homebrew formula built from source and poured from bottle. From `gtkmm/socket.h`:
>> 
>> ```
>> //This file is used only on systems with the X11 backend.
>> //This is available only #ifdef GDK_WINDOWING_X11.
>> //Otherwise this source file will not be compiled,
>> //and no class defined in it will be registered by wrap_init().
>> ```
>> 
>> I don't know much about building software on macOS with X available _at build time_, but it sounds like even less fun than above. The headers are around on my system though – I have XQuartz installed – so I'll give it a whirl.
>> 
>> -- 
> 
> Ah, this can be avoided by only allowing external editors. But, it
> requires some compile-time definitions.
> 

Try out #312 with:

$ scons --disable-embedded-editor

**Comment by gauteh** [comment]
Chris Piro writes on mars 27, 2017 22:46:
> I should also mention that it took me like 15 minutes to get astroid running in (Docker for Mac)[https://docs.docker.com/docker-for-mac/install/]. I started with Ubuntu 16.10, followed your install instructions, then `ssh -YC`'d into the container. This might be the easier path.
> 

Not bad, yeah - that could work as well - I assume you share filesystem
between host and guest OS then?

**Comment by mgraham** [comment]
I'm trying to follow along but I'm new to OSX and homebrew.  I got webkitgtk@2.4.11.rb installed (thanks @cpiro!), but now when installling astroid I'm getting this: 
```
compiling ==> src/command_bar.cc
compiling ==> src/compose_message.cc
compiling ==> src/config.cc
compiling ==> src/crypto.cc
compiling ==> src/db.cc
compiling ==> src/main_window.cc
compiling ==> src/message_thread.cc
compiling ==> src/poll.cc
compiling ==> src/modes/edit_message.cc
src/modes/edit_message.cc:9:11: fatal error: 'gdk/gdkx.h' file not found
# include <gdk/gdkx.h>
```

**Comment by yeled** [comment]
yes, I get that too

**Comment by cpiro** [comment]
Unfortunately the gtk in Homebrew builds with the Quartz backend only, so the X-specific bits won't work. idk how much more work it would be to carve all that out and use backend-independent code.

**Comment by mgraham** [comment]
Ok, well, I'm in over my head, then.  I'm going to try the Docker approach

**Comment by yeled** [comment]
try this `Dockerfile` and run it with `ip=$(ifconfig en0 | grep inet | awk '$1=="inet" {print $2}') && sudo docker run -e HOME="/Users/$USER" -e DISPLAY=$ip:0 -v /tmp/.X11-unix:/tmp/.X11-unix -v /Users/$USER/:/Users/$USER  astroid` from an _XQuartz terminal_.


```
FROM debian:stretch
MAINTAINER Charlie Allom <charlie@evilforbeginners.com>

ENV LANG C.UTF-8

RUN apt-get update
RUN apt-get install -y \
        scons \
        libnotmuch-dev \
        libwebkitgtk-dev \
        g++
RUN apt-get install -y \
        libglibmm-2.4-dev \
        libgtkmm-3.0-dev \
        libgmime-2.6-dev \
        libsass-dev
RUN apt-get install -y \
        libboost1.62-dev \
        libwebkitgtk-3.0-dev scons\
        gobject-introspection \
        git-core
RUN apt-get install -y \
        libgirepository1.0-dev \
        libboost-filesystem-dev \
        libpeas-dev
RUN apt-get clean && \
    rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*

RUN git clone https://github.com/gauteh/astroid.git && cd astroid && scons
RUN ./astroid

VOLUME /Users
WORKDIR /Users
```

**Comment by gauteh** [comment]
Michael Graham writes on april 12, 2017 19:10:
> I'm trying to follow along but I'm new to OSX and homebrew.  I got webkitgtk@2.4.11.rb installed (thanks @cpiro!), but now when installling astroid I'm getting this: 
> ```
> compiling ==> src/command_bar.cc
> compiling ==> src/compose_message.cc
> compiling ==> src/config.cc
> compiling ==> src/crypto.cc
> compiling ==> src/db.cc
> compiling ==> src/main_window.cc
> compiling ==> src/message_thread.cc
> compiling ==> src/poll.cc
> compiling ==> src/modes/edit_message.cc
> src/modes/edit_message.cc:9:11: fatal error: 'gdk/gdkx.h' file not found
> # include <gdk/gdkx.h>
> ```

This should be fixed in latest master (did not need that include),
though you might need to build with:

$ scons --disable-embedded-editor

**Comment by yeled** [comment]
that's funny, I thought I tried that.

Here is the output, OSX dies on linking now: https://gist.github.com/yeled/65297d55c88e60ddc267abff33753638

(I had to edit `/usr/local/Homebrew/Library/Homebrew/download_strategy.rb` in to stop it fetching the git submodules like ssh://aur@aur.archlinux.org/astroid.git which failed of course)

```
 Library/Homebrew/download_strategy.rb | 1 -
 1 file changed, 1 deletion(-)

diff --git Library/Homebrew/download_strategy.rb Library/Homebrew/download_strategy.rb
index a0025cef3f..40746627e0 100644
--- Library/Homebrew/download_strategy.rb
+++ Library/Homebrew/download_strategy.rb
@@ -837,7 +837,6 @@ class GitDownloadStrategy < VCSDownloadStrategy
     cached_location.cd do
       safe_system "git", "config", "homebrew.cacheversion", cache_version
       checkout
-      update_submodules if submodules?
     end
   end
 
```

**Comment by gauteh** [comment]
Charlie Allom writes on april 18, 2017 9:18:
> that's funny, I thought I tried that.

I made some small changes. Try with `--disable-plugins` as well, seems
like there's an GIR issue.

**Comment by yeled** [comment]
I've updated the gist with a new build error. I have to go AFK for a few hours now.

**Comment by gauteh** [comment]
Charlie Allom writes on april 18, 2017 9:42:
> I've updated the gist with a new build error. I have to go AFK for a few hours now.
> 

Hm - `boost::logging` again. Is it the latest version? 


Unavailabe for a while as well, Gaute

**Comment by yeled** [comment]
Yes, latest.
```
11:22 YELP-CHARLIE~[local@2.7.13] % brew info boost
boost: stable 1.63.0 (bottled), HEAD
Collection of portable C++ source libraries
...
```

**Comment by yeled** [comment]
I see misleading stackoverflow articles that recommend linking `-lboost_thread` and such. Nothing of it is working for me though - do you have advice about that?

**Comment by gauteh** [comment]
Charlie Allom writes on april 18, 2017 16:21:
> I see misleading stackoverflow articles that recommend linking `-lboost_thread` and such. Nothing of it is working for me though - do you have advice about that?
> 

Can you send the output of `scons --verbose`? I just did one run on my
system which you can use for comparison: http://ix.io/r4o

**Comment by yeled** [comment]
I can make it link now, with `-lboost_log-mt`

```
/usr/bin/clang++ -o astroid -L/usr/local/opt/icu4c/lib -L/usr/local/opt/openssl/lib -L/usr/local/opt/sqlite/lib -L/usr/local/opt/readline/lib -L/usr/local/opt/libffi/lib -L/usr/local/opt/gettext/lib -L/usr/local/lib -F/usr/local/Frameworks -Wl,-headerpad_max_install_names -lboost_log-mt -lboost_thread-mt src/main.o src/account_manager.o src/astroid.o src/chunk.o src/command_bar.o src/compose_message.o src/config.o src/crypto.o src/db.o src/main_window.o src/message_thread.o src/poll.o src/modes/edit_message.o src/modes/forward_message.o src/modes/help_mode.o src/modes/keybindings.o src/modes/log_view.o src/modes/mode.o src/modes/paned_mode.o src/modes/raw_message.o src/modes/reply_message.o src/modes/saved_searches.o src/modes/thread_index/query_loader.o src/modes/thread_index/thread_index.o src/modes/thread_index/thread_index_list_cell_renderer.o src/modes/thread_index/thread_index_list_view.o src/modes/thread_view/dom_utils.o src/modes/thread_view/theme.o src/modes/thread_view/thread_view.o src/modes/thread_view/web_inspector.o src/modes/editor/editor.o src/modes/editor/external.o src/modes/editor/plugin.o src/actions/action.o src/actions/action_manager.o src/actions/cmdaction.o src/actions/difftag_action.o src/actions/onmessage.o src/actions/tag_action.o src/actions/toggle_action.o src/utils/address.o src/utils/cmd.o src/utils/date_utils.o src/utils/gravatar.o src/utils/resource.o src/utils/ustring_utils.o src/utils/utils.o src/utils/vector_utils.o src/utils/gmime/gmime-filter-html-bq.o src/utils/gmime/gtrie.o src/utils/gmime/url-scanner.o -L/usr/local/Cellar/glibmm/2.50.1/lib -L/usr/local/Cellar/glib/2.52.0/lib -L/usr/local/opt/gettext/lib -L/usr/local/Cellar/libsigc++/2.10.0/lib -L/usr/local/Cellar/gtkmm3/3.22.0/lib -L/usr/local/Cellar/atkmm/2.24.2/lib -L/usr/local/Cellar/pangomm/2.40.1/lib -L/usr/local/Cellar/gtk+3/3.22.11/lib -L/usr/local/Cellar/pango/1.40.5/lib -L/usr/local/Cellar/atk/2.24.0/lib -L/usr/local/Cellar/cairo/1.14.8/lib -L/usr/local/Cellar/cairomm/1.12.2/lib -L/usr/local/Cellar/gdk-pixbuf/2.36.6/lib -L/usr/local/Cellar/gmime/2.6.23/lib -L/usr/local/Cellar/webkitgtk@2.4.11/2.4.11/lib -L/usr/local/Cellar/libsoup/2.56.0/lib -L/usr/local/Cellar/libsass/3.4.3/lib -L/usr/local/Cellar/vte3/0.48.2/lib -L/usr/local/Cellar/pcre2/10.23/lib -L/usr/local/Cellar/gnutls/3.5.11/lib -lsass -lnotmuch -lgtkmm-3.0 -latkmm-1.6 -lgdkmm-3.0 -lgiomm-2.4 -lpangomm-1.4 -lglibmm-2.4 -lcairomm-1.0 -lsigc-2.0 -lgmime-2.6 -lwebkitgtk-3.0 -lsoup-2.4 -ljavascriptcoregtk-3.0 -lsass -lvte-2.91 -lgtk-3 -lgdk-3 -lpangocairo-1.0 -lpango-1.0 -latk-1.0 -lcairo-gobject -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0 -lintl -lz -lpcre2-8 -lgnutls -lboost_filesystem -lboost_system -lboost_program_options -lboost_log_setup -lboost_log -lboost_thread-mt -lboost_date_time -lstdc++
Install file: "astroid" as "/usr/local/bin/astroid"
Install file: "ui/about.glade" as "/usr/local/share/astroid/ui/about.glade"
Install file: "ui/edit-message.glade" as "/usr/local/share/astroid/ui/edit-message.glade"
Install file: "ui/no-mail.png" as "/usr/local/share/astroid/ui/no-mail.png"
Install file: "ui/thread-view.html" as "/usr/local/share/astroid/ui/thread-view.html"
Install file: "ui/icons/LICENSE" as "/usr/local/share/astroid/ui/icons/LICENSE"
Install file: "ui/icons/horizontal_color.png" as "/usr/local/share/astroid/ui/icons/horizontal_color.png"
Install file: "ui/icons/horizontal_color.svg" as "/usr/local/share/astroid/ui/icons/horizontal_color.svg"
Install file: "ui/icons/horizontal_white.png" as "/usr/local/share/astroid/ui/icons/horizontal_white.png"
Install file: "ui/icons/horizontal_white.svg" as "/usr/local/share/astroid/ui/icons/horizontal_white.svg"
Install file: "ui/icons/icon_color.png" as "/usr/local/share/astroid/ui/icons/icon_color.png"
Install file: "ui/icons/icon_color.svg" as "/usr/local/share/astroid/ui/icons/icon_color.svg"
Install file: "ui/icons/icon_white.png" as "/usr/local/share/astroid/ui/icons/icon_white.png"
Install file: "ui/icons/icon_white.svg" as "/usr/local/share/astroid/ui/icons/icon_white.svg"
Install file: "ui/icons/vertical_color.png" as "/usr/local/share/astroid/ui/icons/vertical_color.png"
Install file: "ui/icons/vertical_color.svg" as "/usr/local/share/astroid/ui/icons/vertical_color.svg"
Install file: "ui/icons/vertical_white.png" as "/usr/local/share/astroid/ui/icons/vertical_white.png"
Install file: "ui/icons/vertical_white.svg" as "/usr/local/share/astroid/ui/icons/vertical_white.svg"
Install file: "ui/icons/icon_color.png" as "/usr/local/share/icons/hicolor/512x512/apps/astroid.png"
Install file: "ui/icons/icon_color.svg" as "/usr/local/share/icons/hicolor/scalable/apps/astroid.svg"
Install file: "ui/thread-view.scss" as "/usr/local/share/astroid/ui/thread-view.scss"
Install file: "ui/astroid.desktop" as "/usr/local/share/applications/astroid.desktop"
```

however, the linking isn't right:

```
10:16 YELP-CHARLIE~[local@2.7.13] % /usr/local/bin/astroid
dyld: Symbol not found: __ZN3JSC16JSCallbackObjectINS_18JSAPIWrapperObjectEE6s_infoE
  Referenced from: /usr/local/opt/webkitgtk@2.4.11/lib/libjavascriptcoregtk-3.0.0.dylib
  Expected in: flat namespace
 in /usr/local/opt/webkitgtk@2.4.11/lib/libjavascriptcoregtk-3.0.0.dylib
zsh: abort      /usr/local/bin/astroid
10:16 YELP-CHARLIE~[local@2.7.13] %
```

**Comment by gauteh** [comment]
Hm, looks like there is something wrong with the webkitgtk build.. doesn't even seem like it is something astroid is using

**Comment by yeled** [comment]
I'm trying to rebuild the webkitgtk but it takes an age..

**Comment by yeled** [comment]
ping @cpiro for help. Does that linking problem look familiar? I don't see anything in `--configure` that looks awry. I don't really know what to look for though, this is my first GTK problem ever.

**Comment by cpiro** [comment]
Looks like that symbol `JSCallbackObject<JSAPIWrapperObject>::s_info` should be defined in some ObjC++ code ([JSAPIWrapperObject.mm](https://opensource.apple.com/source/JavaScriptCore/JavaScriptCore-7601.7.2/API/JSAPIWrapperObject.mm.auto.html)) that isn't distributed with our webkitgtk. We have the header, though, and when `JSC_OBJC_API_ENABLED` is `#define`d (which it must be for all of us), there are references to that missing symbol that fail at runtime. Long story short: we don't need the ObjC{,++} interface, so I make sure `JSC_OBJC_API_ENABLED` is not defined. And I got astroid to run! 

I updated my formula gists: 
- [webkitgtk@2.4.11.rb](https://gist.github.com/cpiro/0ea18227b251b26f94c645dd62b3f9a7) (undefine `JSC_OBJC_API_ENABLED`)
- [astroid.rb](https://gist.github.com/cpiro/dcc8ae89322250bf9379201b30a6a377) (with everyone's fixes from above: `boost_log-mt`, `--disable-*`, `--prefix`)

**Comment by gauteh** [comment]
Chris Piro writes on april 26, 2017 7:25:
> Looks like that symbol `JSCallbackObject<JSAPIWrapperObject>::s_info` should be defined in some ObjC++ code ([JSAPIWrapperObject.mm](https://opensource.apple.com/source/JavaScriptCore/JavaScriptCore-7601.7.2/API/JSAPIWrapperObject.mm.auto.html)) that isn't distributed with our webkitgtk. We have the header, though, and when `JSC_OBJC_API_ENABLED` is `#define`d (which it must be for all of us), there are references to that missing symbol that fail at runtime. Long story short: we don't need the ObjC{,++} interface, so I make sure `JSC_OBJC_API_ENABLED` is not defined. And I got astroid to run! 
> 
> I updated my formula gists: 
> - [webkitgtk@2.4.11.rb](https://gist.github.com/cpiro/0ea18227b251b26f94c645dd62b3f9a7) (undefine `JSC_OBJC_API_ENABLED`)
> - [astroid.rb](https://gist.github.com/cpiro/dcc8ae89322250bf9379201b30a6a377) (with everyone's fixes from above: `boost_log-mt`, `--disable-*`, `--prefix`)
> 

Excellent! That is really great news! Is there something that should be
included in the official SConscript? Will this be available in some
official homebrew repo?

**Comment by yeled** [comment]
Let's setup a Tap that is hosted on the `astroid` github repo?

ps. webkitgtk needs `depends_on "webp"` and astroid can do with `depends_on
"vte3"` as well. Not sure if the latter makes a difference though.

I can help with the Tap today.

**Comment by gauteh** [comment]
Charlie Allom writes on april 26, 2017 9:10:
> Let's setup a Tap that is hosted on the `astroid` github repo?
> 
> ps. webkitgtk needs `depends_on "webp"` and astroid can do with `depends_on
> "vte3"` as well. Not sure if the latter makes a difference though.
> 
> I can help with the Tap today.

Good, I added you as a member to the organization so that you should be
able to create the repository.

**Comment by hugoroy** [comment]
So good to see progress on this to have moar users and contributions! :-)

Thanks guys!

Le 26 avril 2017 09:12:18 GMT+02:00, Gaute Hope <notifications@github.com> a écrit :
>Charlie Allom writes on april 26, 2017 9:10:
>> Let's setup a Tap that is hosted on the `astroid` github repo?
>> 
>> ps. webkitgtk needs `depends_on "webp"` and astroid can do with
>`depends_on
>> "vte3"` as well. Not sure if the latter makes a difference though.
>> 
>> I can help with the Tap today.
>
>Good, I added you as a member to the organization so that you should be
>able to create the repository.

-- 
Envoyé de mon appareil Android avec K-9 Mail. Veuillez excuser ma brièveté.

**Comment by yeled** [comment]
@gauteh can you please cut a new release with the changes you made for this issue, and then update the new homebrew [formula](https://github.com/astroidmail/homebrew-astroid/blob/master/Formula/astroid.rb) with the version and sha256.

It *should* be as simple as
```
14:22 YELP-CHARLIE~[local@2.7.13] % brew tap astroidmail/astroid
==> Tapping astroidmail/astroid
Cloning into '/usr/local/Homebrew/Library/Taps/astroidmail/homebrew-astroid'...
remote: Counting objects: 5, done.
remote: Compressing objects: 100% (4/4), done.
remote: Total 5 (delta 0), reused 4 (delta 0), pack-reused 0
Unpacking objects: 100% (5/5), done.
Tapped 2 formulae (29 files, 25.6KB)
14:22 YELP-CHARLIE~[local@2.7.13] % brew install astroid
...
```

once you do this.

**Comment by gauteh** [comment]
Yes, I'll cut a new realease ASAP.

**Comment by gauteh** [comment]
I'm really happy to close this issue! Thanks for all the help!

**Comment by mgraham** [comment]
So I'm finally picking this up again (I had to drop it because I'm new to notmuchmail, and I had to free up a few GB for mail, figure out offlineimap, etc.).  Anyway, I'm trying again to get astroid to run via homebrew.

I've installed astroid 0.9.1 via homebrew.  When I run it, I get the following:
```
~ ❯❯❯ astroid                                                                                                                       ⏎
[11:26:46.908663] [0x00007fff7ac58000] [info] welcome to astroid! - v0.9.1
[11:26:46.908953] [0x00007fff7ac58000] [debug] utf8: true, UTF-8
[11:26:46.971650] [0x00007fff7ac58000] [debug] HOME: /Users/michael
[11:26:46.971757] [0x00007fff7ac58000] [info] cf: loading: "/Users/michael/.config/astroid/config"
[11:26:46.972945] [0x00007fff7ac58000] [info] cf: version: 7
[11:26:46.973609] [0x00007fff7ac58000] [info] notmuch db: /Users/michael/.config/mail
[11:26:46.973706] [0x00007fff7ac58000] [info] date: init.
[11:26:46.973799] [0x00007fff7ac58000] [info] db path: /Users/michael/.config/mail
[11:26:46.974156] [0x00007fff7ac58000] [debug] searches: loading history..
[11:26:46.974285] [0x00007fff7ac58000] [info] ac: initializing accounts..
[11:26:46.974419] [0x00007fff7ac58000] [info] ac: setup account: charlie for Charlie Root (default: true)
[11:26:46.974596] [0x00007fff7ac58000] [info] global actions: set up.
[11:26:46.974767] [0x00007fff7ac58000] [info] poll: setting up.
[11:26:46.974875] [0x00007fff7ac58000] [debug] poll: interval: 60
[11:26:46.974994] [0x00007fff7ac58000] [debug] poll: requested..
[11:26:46.975114] [0x00007fff7ac58000] [info] db: open db read-only, waiting for lock..
[11:26:46.975239] [0x00007fff7ac58000] [debug] db: read-only got lock.

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeCryptoContext'

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeDecryptResult'

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeCertificateList'

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeSignatureList'

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeCertificate'

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeSignature'

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeCryptoContext'

(<unknown>:31116): GLib-GObject-CRITICAL **: g_type_register_static: assertion 'parent_type > 0' failed

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeCryptoContext'

(<unknown>:31116): GLib-GObject-CRITICAL **: g_type_register_static: assertion 'parent_type > 0' failed

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeFilter'

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeFilter'

(<unknown>:31116): GLib-GObject-CRITICAL **: g_type_register_static: assertion 'parent_type > 0' failed

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeFilter'

(<unknown>:31116): GLib-GObject-CRITICAL **: g_type_register_static: assertion 'parent_type > 0' failed

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeFilter'

(<unknown>:31116): GLib-GObject-CRITICAL **: g_type_register_static: assertion 'parent_type > 0' failed

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeFilter'

(<unknown>:31116): GLib-GObject-CRITICAL **: g_type_register_static: assertion 'parent_type > 0' failed

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeFilter'

(<unknown>:31116): GLib-GObject-CRITICAL **: g_type_register_static: assertion 'parent_type > 0' failed

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeFilter'

(<unknown>:31116): GLib-GObject-CRITICAL **: g_type_register_static: assertion 'parent_type > 0' failed

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeFilter'

(<unknown>:31116): GLib-GObject-CRITICAL **: g_type_register_static: assertion 'parent_type > 0' failed

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeFilter'

(<unknown>:31116): GLib-GObject-CRITICAL **: g_type_register_static: assertion 'parent_type > 0' failed

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeFilter'

(<unknown>:31116): GLib-GObject-CRITICAL **: g_type_register_static: assertion 'parent_type > 0' failed

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeFilter'

(<unknown>:31116): GLib-GObject-CRITICAL **: g_type_register_static: assertion 'parent_type > 0' failed

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeFilter'

(<unknown>:31116): GLib-GObject-CRITICAL **: g_type_register_static: assertion 'parent_type > 0' failed

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeFilter'

(<unknown>:31116): GLib-GObject-CRITICAL **: g_type_register_static: assertion 'parent_type > 0' failed

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeStream'

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeStream'

(<unknown>:31116): GLib-GObject-CRITICAL **: g_type_register_static: assertion 'parent_type > 0' failed

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeStream'

(<unknown>:31116): GLib-GObject-CRITICAL **: g_type_register_static: assertion 'parent_type > 0' failed

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeStream'

(<unknown>:31116): GLib-GObject-CRITICAL **: g_type_register_static: assertion 'parent_type > 0' failed

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeStream'

(<unknown>:31116): GLib-GObject-CRITICAL **: g_type_register_static: assertion 'parent_type > 0' failed

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeStream'

(<unknown>:31116): GLib-GObject-CRITICAL **: g_type_register_static: assertion 'parent_type > 0' failed

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeStream'

(<unknown>:31116): GLib-GObject-CRITICAL **: g_type_register_static: assertion 'parent_type > 0' failed

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeStream'

(<unknown>:31116): GLib-GObject-CRITICAL **: g_type_register_static: assertion 'parent_type > 0' failed

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeStream'

(<unknown>:31116): GLib-GObject-CRITICAL **: g_type_register_static: assertion 'parent_type > 0' failed

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeStream'

(<unknown>:31116): GLib-GObject-CRITICAL **: g_type_register_static: assertion 'parent_type > 0' failed

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeStream'

(<unknown>:31116): GLib-GObject-CRITICAL **: g_type_register_static: assertion 'parent_type > 0' failed

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeParser'

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeObject'

(<unknown>:31116): GLib-GObject-CRITICAL **: g_type_register_static: assertion 'parent_type > 0' failed

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeDataWrapper'

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeContentType'

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeContentDisposition'

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'InternetAddress'

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'InternetAddressList'

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'InternetAddress'

(<unknown>:31116): GLib-GObject-CRITICAL **: g_type_register_static: assertion 'parent_type > 0' failed

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'InternetAddress'

(<unknown>:31116): GLib-GObject-CRITICAL **: g_type_register_static: assertion 'parent_type > 0' failed

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeObject'

(<unknown>:31116): GLib-GObject-CRITICAL **: g_type_register_static: assertion 'parent_type > 0' failed

(<unknown>:31116): gmime-CRITICAL **: g_mime_object_register_type: assertion 'object_type != 0' failed

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeObject'

(<unknown>:31116): GLib-GObject-CRITICAL **: g_type_register_static: assertion 'parent_type > 0' failed

(<unknown>:31116): gmime-CRITICAL **: g_mime_object_register_type: assertion 'object_type != 0' failed

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeObject'

(<unknown>:31116): GLib-GObject-CRITICAL **: g_type_register_static: assertion 'parent_type > 0' failed

(<unknown>:31116): GLib-GObject-CRITICAL **: g_type_register_static: assertion 'parent_type > 0' failed

(<unknown>:31116): gmime-CRITICAL **: g_mime_object_register_type: assertion 'object_type != 0' failed

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeObject'

(<unknown>:31116): GLib-GObject-CRITICAL **: g_type_register_static: assertion 'parent_type > 0' failed

(<unknown>:31116): GLib-GObject-CRITICAL **: g_type_register_static: assertion 'parent_type > 0' failed

(<unknown>:31116): gmime-CRITICAL **: g_mime_object_register_type: assertion 'object_type != 0' failed

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeObject'

(<unknown>:31116): GLib-GObject-CRITICAL **: g_type_register_static: assertion 'parent_type > 0' failed

(<unknown>:31116): gmime-CRITICAL **: g_mime_object_register_type: assertion 'object_type != 0' failed

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeObject'

(<unknown>:31116): GLib-GObject-CRITICAL **: g_type_register_static: assertion 'parent_type > 0' failed

(<unknown>:31116): gmime-CRITICAL **: g_mime_object_register_type: assertion 'object_type != 0' failed

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeObject'

(<unknown>:31116): GLib-GObject-CRITICAL **: g_type_register_static: assertion 'parent_type > 0' failed

(<unknown>:31116): gmime-CRITICAL **: g_mime_object_register_type: assertion 'object_type != 0' failed

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'GMimeObject'

(<unknown>:31116): GLib-GObject-CRITICAL **: g_type_register_static: assertion 'parent_type > 0' failed

(<unknown>:31116): GLib-GObject-CRITICAL **: g_type_register_static: assertion 'parent_type > 0' failed

(<unknown>:31116): gmime-CRITICAL **: g_mime_object_register_type: assertion 'object_type != 0' failed
[11:26:46.984459] [0x00007fff7ac58000] [debug] db: open time: 1.997 ms.
[11:26:46.984556] [0x00007fff7ac58000] [info] db: closing db.
[11:26:46.984669] [0x00007fff7ac58000] [debug] db: ro: waiting for lock to close..
[11:26:46.984741] [0x00007fff7ac58000] [debug] db: ro: closing..
[11:26:46.984802] [0x00007fff7ac58000] [debug] poll: revision before poll: 84909
[11:26:46.984945] [0x0000700000323000] [info] poll: polling: /Users/michael/.config/astroid/poll.sh
[11:26:46.985103] [0x0000700000323000] [error] poll: poll script does not exist or is not a regular file.
[11:26:46.985219] [0x00007fff7ac58000] [warning] astroid: starting a new window..

(<unknown>:31116): Gdk-WARNING **: GdkQuartzDisplay does not implement the monitor vfuncs
[11:26:47.198039] [0x00007fff7ac58000] [info] db: open db read-only, waiting for lock..
[11:26:47.198150] [0x00007fff7ac58000] [debug] db: read-only got lock.
[11:26:47.198503] [0x00007fff7ac58000] [debug] db: open time: 0.471 ms.
[11:26:47.206073] [0x00007fff7ac58000] [info] db: loaded 7 tags.
[11:26:47.217647] [0x00007fff7ac58000] [info] db: closing db.
[11:26:47.217873] [0x00007fff7ac58000] [debug] db: ro: waiting for lock to close..
[11:26:47.218049] [0x00007fff7ac58000] [debug] db: ro: closing..
[11:26:47.218161] [0x00007fff7ac58000] [debug] mw: init, id: 1
[11:26:47.218335] [0x00007fff7ac58000] [info] re: using installed resource: /usr/local/Cellar/astroid/0.9.1/share/astroid/ui/icons/icon_color.png
[11:26:47.237059] [0x00007fff7ac58000] [info] astroid: got query: inbox: tag:inbox
[11:26:47.241907] [0x00007fff7ac58000] [info] re: using installed resource: /usr/local/Cellar/astroid/0.9.1/share/astroid/ui/no-mail.png
[11:26:47.245568] [0x00007fff7ac58000] [debug] pm: add pane
[11:26:47.245926] [0x00007000003a6000] [info] db: open db read-only, waiting for lock..
[11:26:47.246073] [0x00007000003a6000] [debug] db: read-only got lock.
[11:26:47.246494] [0x00007000003a6000] [debug] db: open time: 0.857 ms.
[11:26:47.246622] [0x00007000003a6000] [debug] ql: refresh stats..
[11:26:47.247380] [0x00007fff7ac58000] [debug] mw: does not have focus, will not grab modal.
[11:26:47.247535] [0x00007fff7ac58000] [debug] mw: does not have focus, will not grab modal.
[11:26:47.247695] [0x00007fff7ac58000] [debug] mw: does not have focus, will not grab modal.

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'InternetAddressList'

(<unknown>:31116): GLib-GObject-CRITICAL **: g_object_newv: assertion 'G_TYPE_IS_OBJECT (object_type)' failed

(<unknown>:31116): GLib-GObject-WARNING **: cannot register existing type 'InternetAddress'

(<unknown>:31116): GLib-GObject-CRITICAL **: g_type_register_static: assertion 'parent_type > 0' failed

(<unknown>:31116): GLib-GObject-CRITICAL **: g_object_newv: assertion 'G_TYPE_IS_OBJECT (object_type)' failed
[1]    31116 segmentation fault  astroid
```

**Comment by gauteh** [comment]
I think @yeled had the same issue.

**Comment by yeled** [comment]
This is to do with gmime 3.0.0 and should be raised as a new issue, @mgraham so I can help you there.

**Comment by gauteh** [comment]
Charlie Allom writes on juli 4, 2017 23:47:
> This is to do with gmime 3.0.0 and should be raised as a new issue, @mgraham so I can help you there.

Please raise the issue at the homebrew repo: https://github.com/astroidmail/homebrew-astroid


- gaute

**Comment by mgraham** [comment]
Thanks!  I've opened https://github.com/astroidmail/homebrew-astroid/issues/10
